### PR TITLE
Add a read/write roundtrip Spark integration test for a CRAM and reference on HDFS.

### PR DESCRIPTION
**Pull request by cmnbroad**
This just adds a test to verify that with the issues causing https://github.com/broadinstitute/gatk/issues/2382 (which included both the issues fixed by https://github.com/broadinstitute/gatk/pull/6517 ,and some other issues) are all fixed now.

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java [hunk 1]
```diff
@@ -67,6 +74,44 @@ public void testGCSInputsAndOutputsWithSparkNio(final String gcsInput, final Str
...
+        final GATKPathSpecifier testRefDict = new GATKPathSpecifier(getTestFile("count_reads.dict").getAbsolutePath());
+        final GATKPathSpecifier testRefIndex = new GATKPathSpecifier(getTestFile("count_reads.fasta.fai").getAbsolutePath());
+
+        MiniClusterUtils.runOnIsolatedMiniCluster(cluster -> {
+            final org.apache.hadoop.fs.Path workingDirectory = MiniClusterUtils.getWorkingDir(cluster);
+
+            // copy all the test files to HDFS
+            final org.apache.hadoop.fs.Path cramHDFSPath = new org.apache.hadoop.fs.Path(workingDirectory,"count_reads.cram");
+            final org.apache.hadoop.fs.Path refHDFSPath = new org.apache.hadoop.fs.Path(workingDirectory, "count_reads.fasta");
+            final org.apache.hadoop.fs.Path refHDFSDictPath = new org.apache.hadoop.fs.Path(workingDirectory,"count_reads.dict");
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java:90]: I sort of which java had lightweight type aliases so we could rename this locally to HdfsPath and avoid these nasty fully specified names...

File: src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java [hunk 2]
```diff
@@ -67,6 +74,44 @@ public void testGCSInputsAndOutputsWithSparkNio(final String gcsInput, final Str
...
+            final org.apache.hadoop.fs.Path refHDFSIndexPath = new org.apache.hadoop.fs.Path(workingDirectory, "count_reads.fasta.fai");
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testCram.getURI()), cramHDFSPath);
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testRef.getURI()), refHDFSPath);
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testRefDict.getURI()), refHDFSDictPath);
+            cluster.getFileSystem().copyFromLocalFile(new org.apache.hadoop.fs.Path(testRefIndex.getURI()), refHDFSIndexPath);
+
+            // run PrintReadsSpark and print the contents of the HDFS cram test file to an output HDFS cram
+            final GATKPathSpecifier outputHDFSPath = new GATKPathSpecifier(workingDirectory + "testCramOnHDFSOut.cram");
+            final ArgumentsBuilder argBuilder = new ArgumentsBuilder();
+            argBuilder.add("input", cramHDFSPath.toUri().toString())
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java:100]: nitpick, i'm a booster for argBuilder.addInput and .addOutput, but it doesn't matter at all

File: src/test/resources/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark/count_reads.fasta [hunk 3]
```diff
@@ -0,0 +1,40 @@
+>chr1
```
Comment (lbergelson) [inline src/test/resources/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark/count_reads.fasta:1]: I love this tiny reference :). it's so cute!

File: src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java (hunk 1 above)
Comment (cmnbroad) [reply src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java:90]: Yeah, I frequently wish Java had type aliases.

File: src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java (hunk 2 above)
Comment (cmnbroad) [reply src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java:100]: I keep forgetting about those. Easy enough to change though.

**Comment by cmnbroad** [comment]
Thanks for the quick review @lbergelson!

**Pull request merged**